ckanext/aberdeen/actions.py

```diff
@@ -233,18 +253,68 @@ def send_inactive_users_email(context, data_dict):
     '''Sends the inactive users list to all site system administrators'''
 
     user_list = ckan.logic.get_action('user_list')(context, data_dict)
-    admin_email_list = [
-        user['email'] for user in user_list if user['sysadmin'] is True]
-
+    admin_list = [user for user in user_list if user.get('sysadmin') is True]
     inactive_users = toolkit.get_action('inactive_users')(context, data_dict)
+    number_of_users = len(inactive_users)
 
-    # Format the inactive users list for better readability in an email
-    inactive_users = '\n\n'.join(
-        '\n'.join(info + ': ' + str(user_info[info]) for info in user_info)
-        for user_info in inactive_users)
-
-    for admin_email in admin_email_list:
-        if admin_email:
+    if not inactive_users:
+        inactive_users = \
+            'Hi {},\n\nThere are no inactive users to report this week.'
+    else:
+        inactive_users = \
+            'Hi {},\n\nThere are currently {} inactive users:\n\n'
+
+        # Format the inactive users list for better readability in an email
+        for user in user_list:
+            display_name = user.get('display_name')
+            name = user.get('name')
+            profile_url = '{}/user/{}'.format(
+                config.get('ckan.site_url'), name)
+            id = user.get('id')
+            last_activity = user.get('last_activity')
```

Comment:
I see this is None in some cases. Let's make it equal to created if so

last_activity = user.get('last_activity', user.get('created'))

Reply:
Good catch. That makes sense.

Reply:
So, for this, I actually ended up implementing it in the `inactive_users` call, that way it outputs this way whether you call that endpoint OR it's emailed.

Another thing I just realized, is that `.get()` is only useful if a dictionary key doesn't exist, not if it's only none, so I removed all of the `.get()` uses, because all of the fields I'm using will always exist by default.